internal/controllers/import_controller.go

```diff
@@ -164,36 +159,27 @@ func (r *CAPIImportReconciler) reconcileNormal(ctx context.Context, capiCluster
 			return ctrl.Result{}, nil
 		}
 
-		if err := r.createRancherCluster(ctx, capiCluster); err != nil {
-			log.Error(err, "failed creating rancher cluster")
-			return ctrl.Result{}, err
+		if err := rancherClusterHandler.Create(&rancher.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      rancherClusterNameFromCAPICluster(capiCluster.Name),
+				Namespace: capiCluster.Namespace,
+			},
+		}); err != nil {
+			return ctrl.Result{}, fmt.Errorf("error creating rancher cluster: %w", err)
 		}
-		return ctrl.Result{Requeue: true}, nil
-	}
 
-	// Get the cluster name
-	clusterStatus, ok := rancherCluster.Object["status"]
-	if !ok {
-		log.Info("cluster status not set, requeue")
-		return ctrl.Result{Requeue: true}, nil
-	}
-	clusterName, ok := clusterStatus.(map[string]interface{})["clusterName"]
-	if !ok {
-		log.Info("clusterName not set, requeue")
 		return ctrl.Result{Requeue: true}, nil
 	}
-	log.Info("found cluster name", "name", clusterName)
 
-	agentDeployed, ok := rancherCluster.Object["status"].(map[string]interface{})["agentDeployed"]
-	if ok {
-		if agentDeployed.(bool) {
-			log.Info("agent already deployed, no action needed")
-			return ctrl.Result{}, nil
-		}
+	log.Info("found cluster name", "name", rancherCluster.Status.ClusterName)
```

Comment:
We need to requeue is `clusterName` is blank.  Although the original code only checked for the existence of the field and not if it was non-empty.

Reply:
We could do this in a follow up